cypress/e2e/usecase_page.cy.ts

```diff
@@ -0,0 +1,129 @@
+const API_URL = Cypress.env('apiUrl')
+const TEST_AGENT_URL = 'http://localhost:9000'
+
+describe('UseCase Page', () => {
+  it('successfully completes school use case', () => {
+    cy.visit('/demo')
+    cy.get('[data-cy=next-onboarding-step]').click()
+    cy.get('[data-cy=use-wallet]').first().click()
+    cy.get('[data-cy=small-button]').click()
+
+    cy.get('[data-cy=select-char]').first().click()
+    cy.intercept('POST', `${API_URL}/oob/create-legacy-invitation`).as('createInvitation')
+    cy.get('[data-cy=next-onboarding-step]').click()
+
+    cy.wait('@createInvitation').then((interception) => {
+      const body = { invitationUrl: interception.response?.body.invitationUrl, autoAcceptConnection: true }
+
+      const oobId = interception.response?.body.outOfBandRecord.id
+
+      cy.intercept('GET', `${API_URL}/connections?outOfBandId=${oobId}`).as('getConnectionRecord')
+
+      cy.request('POST', `${TEST_AGENT_URL}/oob/receive-invitation-url`, body)
+
+      cy.wait(['@getConnectionRecord']).then((inter) => {
+        const record = inter.response?.body[0]
+        cy.wrap(record).its('state').should('not.equal', 'invited')
+      })
+    })
+
+    cy.intercept('POST', '/credentials/offer-credential').as('offerCredential')
+
+    cy.get('[data-cy=next-onboarding-step]').click()
+
+    cy.wait('@offerCredential').then((interception) => {
+      const connectionId = interception.response?.body.connectionId
+      const threadId = interception.response?.body.threadId
+
+      cy.request('GET', `${TEST_AGENT_URL}/credentials/`).should((response) => {
+        const testAgentRecord = response.body.find((x) => x.threadId === threadId && x.state === 'offer-received')
```

Comment:
```suggestion
        const credentialRecord = response.body.find((credentialRecord) => credentialRecord.threadId === threadId && credentialRecord.state === 'offer-received')
```

I assume it is a credential record.